array_api_compat/common/_helpers.py

```diff
@@ -77,8 +78,10 @@ def is_array_api_obj(x):
         or hasattr(x, '__array_namespace__')
 
 def _check_api_version(api_version):
-    if api_version is not None and api_version != '2021.12':
-        raise ValueError("Only the 2021.12 version of the array API specification is currently supported")
+    if api_version == '2021.12':
+        warnings.warn("The 2021.12 version of the array API specification was requested but the returned namespace is actually version 2022.12")
+    if api_version is not None and api_version != '2022.12':
+        raise ValueError("Only the 2022.12 version of the array API specification is currently supported")
```

Comment:
The text of the error isn't right, no? Because 2022.12 and 2021.12 are supported

---

Can we somehow re-do the `if` statements? I think something like the below would be easier to read and implement the behaviour Ralf suggested (no warning, just return subset)
```python
if api_version is not in ('2021.12', '2022.12', None):
    raise ValueError(f"The specified Array API version '{api_version}' is not supported.")
```
WDYT?